compiler.pri

```diff
@@ -227,14 +227,44 @@ unix {
 		QMAKE_OBJECTIVE_CXXFLAGS *= -O3 -march=native -ffast-math -ftree-vectorize -fprofile-use
 	}
 
-	CONFIG(c++11) {
-		# Qt 5 will pass the expected compiler flags
-		# needed for C++11 mode when CONFIG includes c++11.
-		# But Qt 4 won't, so we add it manually.
-		lessThan(QT_MAJOR_VERSION, 5) {
-			QMAKE_CXXFLAGS *= -std=c++11
+	# Allow CONFIG(c++11) to enable C++11 mode
+	# when building against Qt 4.
+	lessThan(QT_MAJOR_VERSION, 5):CONFIG(c++11) {
+		QMAKE_CXXFLAGS *= -std=c++11
+	}
+
+	# Detect Qt >= 5.7. Qt 5.7 and above
+	# require C++11 at the very least.
+	#
+	# The syntax of qmake isn't very
+	# expressive, so this check is done
+	# in two parts.
+	#
+	# First, we check if the major version
+	# is greater than 5, since that definitely
+	# means >= 5.7.
+	greaterThan(QT_MAJOR_VERSION, 5) {
+		CONFIG += qt_5_7_or_later
+	}
+	# Instead, if we're using Qt 5, check
+	# that the minor version is greater
+	# than 6 to check for >= 5.7.
+	equal(QT_MAJOR_VERSION, 5) {
```

Comment:
As you said, should be [`equals`](http://doc.qt.io/qt-5/qmake-test-function-reference.html#equals-variablename-value)